api/scpca_portal/models/dataset.py

```diff
@@ -59,3 +61,100 @@ class Meta:
 
     def __str__(self):
         return f"Dataset {self.id}"
+
+    @property
+    def is_data_valid(self) -> bool:
+        data_validator = DataValidator(self.data)
+        return data_validator.is_valid
+
+
+class DataValidator:
+    def __init__(self, data: Dict[str, Dict]) -> None:
+        self.data: Dict[str, Dict[str, Any]] = data
+
+    @property
+    def is_valid(self) -> bool:
+        return all(self.validate_project(project_id) for project_id in self.data.keys())
+
+    @property
+    def valid_projects(self) -> List[str]:
+        return [project_id for project_id in self.data.keys() if self.validate_project(project_id)]
+
+    @property
+    def invalid_projects(self) -> List[str]:
+        return [
+            project_id for project_id in self.data.keys() if not self.validate_project(project_id)
+        ]
+
+    def validate_project(self, project_id: str) -> bool:
+        if not self._validate_project_id(project_id):
+            return False
+
+        if not self._validate_merge_single_cell(project_id):
+            return False
+
+        if not self._validate_includes_bulk(project_id):
+            return False
+
+        if not self._validate_single_cell(project_id):
+            return False
+
+        if not self._validate_spatial(project_id):
+            return False
+
+        return True
+
+    def _validate_project_id(self, project_id):
+        return self._validate_id(project_id, common.PROJECT_ID_PREFIX)
+
+    def _validate_id(self, id: str, prefix: str) -> bool:
+        if not isinstance(id, str):
+            return False
+
+        if not id.startswith(prefix):
+            return False
+
+        id_number = id.removeprefix(prefix)
+        return len(id_number) == 6 and id_number.isdigit()
+
+    def _validate_merge_single_cell(self, project_id) -> bool:
+        if DatasetDataProjectConfig.MERGE_SINGLE_CELL not in self.data.get(project_id, {}):
+            return True
+
+        return isinstance(
+            self.data.get(project_id).get(DatasetDataProjectConfig.MERGE_SINGLE_CELL, {}), bool
+        )
```

Comment:
```suggestion
    def _validate_merge_single_cell(self, project_id) -> bool:
        if value := self.data.get(project_id, {}).get(DatasetDataProjectConfig.MERGE_SINGLE_CELL):
            return isinstance(value, bool)

        return True
```

Do you think something like this would be easier to read?

Reply:
Look this